cmd/snap-update-ns/update.go

```diff
@@ -39,6 +39,10 @@ type MountProfileUpdateContext interface {
 	LoadCurrentProfile() (*osutil.MountProfile, error)
 	// SaveCurrentProfile saves the mount profile that is currently applied.
 	SaveCurrentProfile(*osutil.MountProfile) error
+	// UID returns the user ID of relevant user
+	UID() int
+	// GID returns the group ID of relevant user
+	GID() int
```

Comment:
as discussed separately this should be attached to Assumptions instead, as the *Context are slightly named in a misleading way, they are not meant to be propagated everywhere, as just a few levels needs to deal with loading/locking/writing, *Scope could have been a better name for them

Reply:
Fixed in replacement: https://github.com/snapcore/snapd/pull/13342/files
I did however simplify by getting the uid where needed in changes.go